engine/servermaster/scheduler/placement_constrainer.go

```diff
@@ -37,11 +37,12 @@ type MockPlacementConstrainer struct {
 
 // GetPlacementConstraint implements PlacementConstrainer.GetPlacementConstraint
 func (c *MockPlacementConstrainer) GetPlacementConstraint(
-	_ context.Context, id resourcemeta.ResourceID,
+	_ context.Context,
+	resourceKey resourcemeta.ResourceKey,
 ) (resourcemeta.ExecutorID, bool, error) {
-	executorID, exists := c.ResourceList[id]
+	executorID, exists := c.ResourceList[resourceKey.ID]
```

Comment:
Can it be conflicts here if two jobs have resources with the same `ID`?

Reply:
Yes, it's for mock and I miss it.